comm/src/main/scala/coop/rchain/comm/PeerNode.scala

```diff
@@ -3,10 +3,12 @@ package coop.rchain.comm
 import java.net.InetSocketAddress
 import com.netaporter.uri.Uri
 import scala.util.control.NonFatal
+import coop.rchain.crypto.codec.Base16
 
 // TODO: Add Show instance
 final case class NodeIdentifier(key: Seq[Byte]) {
-  override def toString: String = key.map("%02x".format(_)).mkString
+  private val keyString         = Base16.encode(key.toArray)
```

Comment:
this field is used only here, why dont keep it inlined as it was before?

Reply:
It's an optimization for performance. We don't want to encode the bytes each time the program calls `toString`.

Reply:
👍